library/src/level1/rocsparse_gather.cpp

```diff
@@ -126,6 +133,13 @@ try
         RETURN_IF_ROCSPARSE_ERROR((rocsparse::gather_template<int64_t, _Float16>)(handle, y, x));
         return rocsparse_status_success;
     }
+    // brain half real ; i64
+    if(x->idx_type == rocsparse_indextype_i64 && x->data_type == rocsparse_datatype_bf16_r)
```

Comment:
Might be the opportunity to refactor the dispatch.

Reply:
Done.